src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private T[] grow(int preferredNewCapacity) {
+        int newCapacity = elementData.length + (elementData.length >> 1);
+        newCapacity = Math.max(preferredNewCapacity, newCapacity);
+        T[] tempElementData = (T[]) new Object[newCapacity];
+        System.arraycopy(elementData, 0, tempElementData, 0, size);
+        elementData = tempElementData;
+        return elementData;
+    }
+
+    private void rangeCheckForAdd(int index) {
```

Comment:
```suggestion
    private void checkRangeForAdd(int index) {
```

method names should sound like nouns. fix all same issues